Shuttle (New) FSC Gaslight

dustylens opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

Oh my gracious is that another one?

This PR adds the FSC Gaslight. 

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

The Gaslight has been on my hypothetical drafting board ever since I put together the Retort. It also came back into focus when I made Helium as a simplified economic entry point into Frontier's unique gas mining system. The Retort was an atmos shuttle made for atmosians. While the Retort is fairly accessible the simple truth is that outlook is informed primarily by understanding its systems and how it can be used as a simple gas collection shuttle and not foundation for atmos reactions. 

For the player looking to get into gas mining conceptually it's a bit of a monster.

The idea was shelved again when the Siphon emerged. But after some consideration, and consultation with our Atmos community, I believe a niche remains.

The Gaslight is designed to perform two functions.

1) Simple gas collection platform with a robust storage capacity system to allow for a long shift of gas collection supported by a simple built in cooling loop.

2) Boast a robust suite of resource extraction equipment to supplement that mission.

The Gaslight leans in fully to the role of Gas Mining as a value added process around asteroid mining with a full salvage kit included. In addition to all of the tools I would want for my own gas mining plans. Including selective gas and cannister fill/empty systems it also has a full built in mining support system to make sure that the shift long process of mining is a rewarding one.

This does result in it being a full double archetype shuttle. Which is, you know, a lot. But in the case of gas mining we ask our shuttles to do a lot.

The Siphon's gas tank runs on a simple full gas input from the blast doors and is supplemented by a selective output system for optional single gas extraction. For the purposes of atmospheric control it uses a series of scrubbers in the gas chamber which can be selected to decide which gasses are no longer desired over a shift. 

## Technical details
<!-- Summary of code changes for easier review. -->

ymaamlaml

## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->

Buy it fly it mine with it.

## Media

<img width="661" height="701" alt="ss+(2025-10-22+at+08 17 59)" src="https://github.com/user-attachments/assets/5e2c3a65-94cb-41d2-b156-93c565b25d7c" />

<img width="104" height="117" alt="ss+(2025-10-20+at+10 14 01)" src="https://github.com/user-attachments/assets/fe4c393f-eec8-474f-a696-ed55f5ce6509" />

<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that the content in this PR is my own work, and/or is properly attributed to the original author(s).
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
🆑 
- add: FSC Gaslight. A new compact gas mining shuttle.

GATE-Elemental commented:
Hello!

I've tested the gaslight, got a couple of enormous oxy, nitrogen and a few muddles.
<img width="593" height="520" alt="image" src="https://github.com/user-attachments/assets/3a8b9326-0a60-4a92-adca-b8fd30e6c351" />

1 - Usage was very straight forward, access to asteroid was easy.

2 - Ship seems to have enough space for a crew of two, workshop feels cramped but has enough space to access all machines with ease.

3 - Storage is fairly big and I assume that with huge luck of finding successive enormous nodes, it would take 6 to 8 nodes to really fill it to 9 mpa.

4 - The cooling is quite slow unless the molage amount is also low, if left unattended it might transform a small muddle into frezon and decomp all frezon into n2o too.

5 - Gas miner would have to access inside the gas storage to turn pumps off in order to start/stop cooling, which will be slightly harder without an atmos suit when pressure is high.

6 - I feel like the ship can be easily used to fill tanks from bigger ships or stations.

I liked it very much.

dustylens commented:
Hello and thank you for the feedback!

Some changes!

- I moved the buttons around for the tank systems. The cooling loop can now be operated via a signal valve. Allowing you to turn it on/off. It is off by default. The on/off status is supported by the valve indicator AND a blue valve.
- A second blast door has been added to the gas storage chamber. The button is now behind a locked frame within view at the front of the tank.
- Welding fuel added. Gotta weld.
- The canister dump port now has a KPA pump.
- The canister fill port now has a volumetric pump.
- ATM now the correct ATM. This is like the third time I've goofed this up.
- Added some bits and bobs. The queen is back. If you know you know.

- Scrap processor removed. 

- The Gaslight is now available at Edison.

GATE-Elemental commented:
Hello again.

I liked the last update, the cooling device is now valved, this is good as we can suggest only cooling when the gas area has a high pressure like over 5 mpa.

I have a small issue with the side blastdoors, their purpose is to quickly vent the room?
I made a holofan and used the locked button, it opened the two blastdoors at the same time, which is good if that is the purpose, however if I need to go in there for whatever reason I would need to set a signaller to open only one of the doors, this should be highlighted in the manual.

Air Alarm in the workshop is accessible via the opened RCD box (or just moved)

I felt very comfortable in this version.

dustylens commented:
> > > Hello again.
> > > I liked the last update, the cooling device is now valved, this is good as we can suggest only cooling when the gas area has a high pressure like over 5 mpa.
> > > I have a small issue with the side blastdoors, their purpose is to quickly vent the room? I made a holofan and used the locked button, it opened the two blastdoors at the same time, which is good if that is the purpose, however if I need to go in there for whatever reason I would need to set a signaller to open only one of the doors, this should be highlighted in the manual.
> > > Air Alarm in the workshop is accessible via the opened RCD box (or just moved)
> > > I felt very comfortable in this version.
> > 
> > 
> > Reasonable! A single blast door wont be fast but you probably don't need to vent the hold in a hurry.
> 
> Probably wont need to visit the gas area if pressure is over 900kpa and I'm out of atmos suits either.

(I usually throw a handful of atmos suits into the reclaimer after a short salvage hop, amusingly)

On the whole I -generally- feel that this shuttle will be a bit more of a "take it as it is. Buy it, turn the key, use it" shuttle. If you know what you're doing you're probably going to spend a little more for a retort and get a full suite of tools. If you're here to grab gas to move it to a refining base or to farm helium to sell shuttes like this and the Siphon are likely more you're speed.

Maybe.

So if you're going to be doing big overhauls to the tank you'll likely be doing them at the start of the shift. But making it so you don't need to juggle double doors is prudent.

dustylens commented:
I think this one isn't in a bad spot. Thank you to all of the people who have field tested it! A big ❤️ to Frontier's atmos crew.

There may be more changes that are requested down the road. But at the moment I think field testing will help shake that out.

the-hivequeen reviewed (approved):
Everything seems in order. Unfortunately, the atmosians have solved this one before it even got out of the gate.
